paymentsheet/src/main/java/com/stripe/android/link/ui/wallet/WalletViewModel.kt

```diff
@@ -78,6 +80,18 @@ internal class WalletViewModel @Inject constructor(
             it.setProcessing()
         }
 
+        viewModelScope.launch {
+            linkAccountManager.consumerPaymentDetails.filterNotNull().collectLatest { consumerPaymentDetails ->
+                if (consumerPaymentDetails.paymentDetails.isEmpty()) {
+                    navigateAndClearStack(LinkScreen.PaymentMethod)
+                } else {
+                    _uiState.update {
+                        it.updateWithResponse(consumerPaymentDetails)
+                    }
+                }
+            }
+        }
```

Comment:
same logic as before, but now we observe payment list changes, that will trigger after fetching them from the backend.